torchtitan/experiments/deepseek_v3/group_gemms.py

```diff
@@ -87,9 +96,65 @@ def is_available() -> bool:
     "DSGroupGEMM",
     "TorchBF16GroupGEMM",
     "TorchAOBF16GroupGEMM",
+    "TritonCGBF16GroupGEMM",
 ]
 
 
+class TritonCGBF16GroupGEMM(GroupGEMMStrategy):
+    """Implementation of Triton Contiguous group Gemm"""
+
+    def arrange_expert_weights(self, all_weights, submod_name, module):
+        """prep the expert weights for group gemm usage"""
+
+        return torch.stack(all_weights)
+
+    def execute(self, contig_tokens, m_sizes, m_offsets, module):
+        # Get weights
+        w_gate = module.get_parameter("gate_proj_weight")
+        w_up = module.get_parameter("up_proj_weight")
+        w_down = module.get_parameter("down_proj_weight")
+
+        # Run first two GEMMs (gate and up projections)
+        # Get only valid tokens
+        valid_tokens = contig_tokens[: m_offsets[-1]]
+
+        # print(f"{len(valid_tokens)=}")
+        # Create indices from offsets without CPU-GPU sync
+        m_indices = dsgemm_utils.create_indices_from_offsets_nosync(m_offsets)
+
+        gate_proj = cg_grouped_gemm_forward(valid_tokens, w_gate, m_indices)
+
+        up_proj = cg_grouped_gemm_forward(valid_tokens, w_up, m_indices)
+
+        # Apply activation
+        hidden_outputs = self.activation_function(gate_proj) * up_proj
+        # print(f"{hidden_outputs=}")
+
+        # Run the third GEMM (down projection)
+        # print(f"valid_tokens.shape={valid_tokens.shape}, dtype={valid_tokens.dtype}")
+        # print(f"w_gate.shape={w_down.shape}, dtype={w_down.dtype}")
+
+        # Create indices from offsets without CPU-GPU sync
+        # m_indices = dsgemm_utils.create_indices_from_offsets_nosync(m_offsets)
+        # print(f"m_indices.shape={hidden_outputs.shape}, m_indices={m_indices[:10]}")
```

Comment:
clean up?

Reply:
yes, thanks for flagging...earlier debugging stuff.